dfm/description/matchcounter.py

```diff
@@ -0,0 +1,129 @@
+from collections import defaultdict
+
+from spacy.language import Language
+from spacy.matcher import Matcher
+from spacy.tokens import Doc
+
+from typing import Dict, Iterable, List, Optional
+
+
+class MatchCounter:
+    """Class for counting matches in a text corpus.
+
+    Args:
+        match_patterns (List[Dict[str, list]): list of lowercase spacy match patterns
+        nlp (Language): The spacy language to use
+    """
+
+    def __init__(self, match_patterns: List[Dict[str, list]], nlp: Language):
+        self.nlp = nlp
+        self.matcher_objects = self.gen_matcher_objects_from_pattern_list(
+            match_patterns
+        )
+
+    @staticmethod
+    def term_list_to_lowercase_match_patterns(
```

Comment:
```suggestion
    def term_list_to_spacy_match_patterns(
```

Given the description of this method, I believe this is a more accurate name? Lowercasing seems like something one might wanna add as a flag to this method.